dbt/scripts/utils/export.py

```diff
@@ -177,54 +177,73 @@ def export_models(
+                    }
 
-                # Apply any column formatting that was configured
+                # Parse any formatting that is configured at the column level.
+                # Note that if formatting is configured for a column that we
+                # parsed as a parid column above, these settings will override
+                # the default parid settings from the block above
                 format_config = model["config"]["meta"].get(
                     "export_format", {}
                 )
                 if column_configs := format_config.get("columns"):
                     for column_config in column_configs:
-                        # Set horizontal alignment if config is present
+                        # The column index is required in order to set any
+                        # column-level configs
+                        col_letter = column_config.get("index")
+                        if col_letter is None:
+                            raise ValueError(
+                                "'index' attribute is required in "
+                                "export_format.columns config for "
+                                f"model {model_name}"
+                            )
+                        idx = column_index_from_string(col_letter) - 1
+                        # Initialize the config dict for this column if
+                        # none exists yet
+                        column_format_by_index[idx] = {}
+                        # Parse configs if they are present
+                        if number_format := column_config.get("number_format"):
+                            column_format_by_index[idx]["number_format"] = (
+                                number_format
+                            )
                         if horiz_align_dir := column_config.get(
                             "horizontal_align"
                         ):
-                            horizontal_alignment = Alignment(
-                                horizontal=horiz_align_dir
+                            column_format_by_index[idx]["alignment"] = (
+                                Alignment(horizontal=horiz_align_dir)
                             )
-                            col_letter = column_config.get("index")
-                            if col_letter is None:
-                                raise ValueError(
-                                    "'index' attribute is required when "
-                                    "'horizontal_align' is set on "
-                                    "export_format.columns config for "
-                                    f"model {model_name}"
-                                )
-                            idx = column_index_from_string(col_letter) - 1
-                            # Skip header row
-                            for row in sheet[2 : sheet.max_row]:
-                                row[idx].alignment = horizontal_alignment
```

Comment:
In contrast to the `parid` formatting above, the previous version of this loop wouldn't have handled single-row spreadsheets properly (i.e. empty sheets with only a header row). We hadn't yet run into a case where this loop needed to format an empty sheet, so it hadn't caused problems yet, but the refactor will handle this case much better if it ever comes up.